zerver/views/__init__.py

```diff
@@ -572,6 +572,36 @@ def dev_direct_login(request, **kwargs):
     return HttpResponseRedirect("%s%s" % (settings.EXTERNAL_URI_SCHEME,
                                           request.get_host()))
 
+@csrf_exempt
+@require_post
+@has_request_variables
+def api_dev_android_direct_login(request, username=REQ):
```

Comment:
What I was saying is when you rebase onto master (which we'll need to do in order to merge this), I think this will stop working.